mslib/msui/msui_mainwindow.py

```diff
@@ -468,6 +473,146 @@ def __init__(self, mscolab_data_dir=None, *args):
             self.actionUpdater.triggered.connect(self.updater.show)
         self.openOperationsGb.hide()
 
+        self.last_changed_item = None
+        self.StoreLayout.clicked.connect(self.store_operation_data)
+        self.RestoreLayout.clicked.connect(self.restore_operation_data)
+        self.allowed_widget_names = ["TopViewWindow", "SideViewWindow", "TableViewWindow", "LinearWindow"]
+        self.listOperationsMSC.itemActivated.connect(self.operation_changed)
+
+    def operation_changed(self):
+        selected_items = self.listOperationsMSC.selectedItems()
+
+        if not selected_items:
+            return
+
+        selected_item = selected_items[0]
+        operation_name = selected_item.text()
+        loaded_data = requests.get(f"{self.mscolab.mscolab_server_url}/load_operation_layout", params={"operation_name": operation_name})
+        if loaded_data.text == "None":
+            return
+
+        if loaded_data is not None:
+            result = QtWidgets.QMessageBox.question(
+                self, "Saved Layout Detected", "Would you like to restore it?",
+                QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No, QtWidgets.QMessageBox.Yes)
+            
+            if result == QtWidgets.QMessageBox.Yes:
+                self.restore_operation_data()
```

Comment:
This will load the data a second time causing additional delays and traffic. Can't you reuse the already retrieved data, e.g. with an optional parameter?